src/server.rs

```diff
@@ -85,7 +85,15 @@ pub async fn run(
     // with rustls, the hyper and tower crates [what axum is built on] had to
     // be used directly
 
-    let config = OrchestratorConfig::load(config_path).await;
+    let config = OrchestratorConfig::load(config_path)
```

Comment:
Can we move the `Orchestrator::load()` call to `main.rs` and provide it to `serve()`, in lieu of `config_path`? As `main()` returns `Result<(), Box<dyn std::error::Error>`, you can then drop the conversion `From<config::Error> for server::Error`.

Suggested change for `main.rs`:
```rust
// ...
 // Launch Tokio runtime
    tokio::runtime::Builder::new_multi_thread()
        .enable_all()
        .build()
        .unwrap()
        .block_on(async {
            let config = OrchestratorConfig::load(args.config_path).await?;
            server::run(
                http_addr,
                health_http_addr,
                args.tls_cert_path,
                args.tls_key_path,
                args.tls_client_ca_cert_path,
                config,
            )
            .await?;
```

Reply:
Really good suggestions thanks, resolved